### PR TITLE
Improve legend error

### PR DESCRIPTION
**Pull request by rubenmoya**
Related to: #1715

The goal was to add an error message related to limits. After some research we realized we can't have limits errors here since the legends come from `viz.json`, it could only fail when instantiating the map or when parsing CartoCSS, and we're already covered.

The only change introduced by this PR is changing the `No data available` styles and refactor the `legend-view-base.js`.

screen shot 2017-09-19 at 10 42 31
screen shot 2017-09-19 at 10 43 30
screen shot 2017-10-03 at 12 06 50

### Acceptance
To show the "Something went wrong" text:

- Create a map with text and numeric values
- Change the color of the points by value and choose a numeric column
- Open the cartocss editor, and change the column name by a text column
- The "Something went wrong" message with a placeholder should appear in the legend

To show the "No data available" text:
- Not sure how we can reproduce it

### CODE REVIEW

File: src/geo/ui/legends/base/legend-view-base.js [hunk 1]
```diff
@@ -21,41 +19,35 @@ var LegendViewBase = Backbone.View.extend({
 
     this._loadImages();
 
-    if (this.model.isVisible()) {
-      this.$el.show();
-    } else {
-      this.$el.hide();
-    }
+    this.model.isVisible()
+      ? this.$el.show()
```
Comment (elenatorro) [inline src/geo/ui/legends/base/legend-view-base.js]: 😍

File: src/geo/ui/legends/base/legend-view-base.js [hunk 2]
```diff
@@ -21,41 +19,35 @@ var LegendViewBase = Backbone.View.extend({
...
-    } else if (this.model.isLoading()) {
-      html.push(this._getPlaceholderHTML());
+    if (this.model.isSuccess() && this.model.isAvailable()) {
+      return this._getLegendHTML();
     }
-    return html.join('\n');
+
+    var placeholder = this._getPlaceholderHTML();
+
+    var error = this.model.isError()
```
Comment (nobuti) [inline src/geo/ui/legends/base/legend-view-base.js]: no loading state?

Comment (rubenmoya) [reply src/geo/ui/legends/base/legend-view-base.js]: Well, that as a fail 😓  let me change it

File: src/geo/ui/legends/base/legend-view-base.js (hunk 1 above)
Comment (nobuti) [reply src/geo/ui/legends/base/legend-view-base.js]: Or `this.$el.toggle(this.model.isVisible())` but I prefer the way you implemented.

**Comment by nobuti** [comment]
Could we prepare it for the acceptance? I guess we will need a builder branch, right?

**Comment by rubenmoya** [comment]
Builder branch created!

**Comment by rubenmoya** [comment]
Deployed to `ded15`

File: src/geo/ui/legends/base/legend-view-base.js (hunk 1 above)
Comment (xavijam) [reply src/geo/ui/legends/base/legend-view-base.js]: @nobuti comment is the winner, IMHO 💃 .

**Comment by xavijam** [comment]
Tested and working 👌 , merging guys!

**Pull request merged**